### ISSUE TITLE
Pinpoint - Duplicate bug posts

### ISSUE THREAD

**Issue by dave-2** [catapult-project/catapult#3872]
Pinpoint is posting multiple duplicate bug comments on completion.
Example: 6 comments on [bug 765641](https://bugs.chromium.org/p/chromium/issues/detail?id=765641#c3).

The job completed at: `2017-09-16 07:13:29.867410 UTC`
The 1st bug comment was at: `Sat Sep 16 07:13:29 2017 UTC`
The 2nd bug comment was at: `Sat Sep 16 07:58:04 2017 UTC`
The last bug comment was at: `Sat Sep 16 07:58:07 2017 UTC`

So one bug comment is being posted at bug completion time, and then, around 45 minutes later, several at once.

My first hunch is that there are duplicate taskqueue tasks, so the job is being completed multiple times. But a search through the log doesn't show any duplicate tasks. The last task was at `2017-09-16 07:13:18.401 UTC` and ran for 11 seconds, which coincides with the first bug comment, so none of the duplicates came from running the job.

The bug comments are posted in `job.Complete()`, which is only called through the `/api/run` handler, and there are no `/api/run/138dbf73b80000` entries in the log after 07:13, so it doesn't seem like other operations, like querying the Job, caused it.

The bug comments are Pinpoint bug comments, so the duplicates are not from dashboard.

So the next two places to look are `issue_tracker_service.py` and Monorail itself.

**Comment by dave-2** [comment]
Sorry, ignore everything in that last comment. Turns out there are **two** Pinpoint tasks on that bug, and the second task _is_ showing duplicate tasks. Then the question is when this happened, and why.

**Comment by dave-2** [comment]
I see calls to `/api/run/14b85b8bb80000` at (all times PDT):
```
1 18:49:16.8 ~ 18:49:19.3
2 18:49:29.2 ~ 18:49:31.5 (10s after #1)
3 18:51:11.9 ~ 18:51:15.4 (100s after #2)
4 18:51:20.9 ~ 18:51:23.1 (5s after #3)
5 18:51:25.3 ~ 18:51:28.5 (10s after #3)
6 18:51:33.0 ~ 18:51:36.0 (10s after #4)
```
So there was some kind of break and then a jump from 1 to 2 tasks.
I know I was messing around with the tasks a bunch yesterday.
This could be either from that, or maybe something from taskqueue?

**Comment by simonhatch** [comment]
The taskqueue retry is set to 2, but I haven't found a call to /api/run/14b85b8bb80000 that posted anything other than 200 yet. Are there any other Pinpoint jobs that posted duplicates? I'll go check the rest and see.

**Comment by dave-2** [comment]
We also saw a bug where the "job failed" comment was posted 3 times. This is because we've set the job taskqueue to 2 retries, and it's being posted for every taskqueue try. I'm disabling retries for now (https://codereview.chromium.org/3017553002/)

I don't think that's the same issue as the duplicate taskqueue entries, since that job had no failures and thus no retries. It's possible that it's because I manually kicked a bunch of jobs on Friday. I'm going to be more hands-off on this week's jobs and watch them carefully for duplicates.
